File: Fortran/caeser.f90
program Caeser
  implicit none 

character(70) :: csrString 
character(70) :: originalMsg
integer :: s

print *, "Type your message: "
read(*,'(A)') csrString
print *, "Enter cipher: "
read(*,'(A)') s

originalMsg = csrString

print *, "Message: ", originalMsg
call encrypt(csrString)
print *, "Encrypted Message: ", csrString
call decrypt(csrString)
print *, "Decrypted Message: ", csrString

contains

subroutine encrypt(message) 
  integer :: i 
  character(*), intent(inout) :: message
  character :: ltr

  do i = 1, LEN_TRIM(message)
    ltr = message(i:i)
    if (ltr >= 'A' .and. ltr <= 'Z') then  
      ltr = achar(modulo(iachar(message(i:i)) - 65 + s, 26) + 65)
      message(i:i) = ltr
    else if (ltr >= 'a' .and. ltr <= 'z') then
      ltr = achar(modulo(iachar(message(i:i)) - 97 + s, 26) + 97)
      message(i:i) = ltr
    end if
  end do
end subroutine 

subroutine decrypt(message) 
  integer :: i 
  character(*), intent(inout) :: message
  character :: ltr

  do i = 1, LEN_TRIM(message)
    ltr = message(i:i)
    if (ltr >= 'A' .and. ltr <= 'Z') then  
      ltr = achar(modulo(iachar(message(i:i)) - 65 - s, 26) + 65)
      message(i:i) = ltr
    else if (ltr >= 'a' .and. ltr <= 'z') then
      ltr = achar(modulo(iachar(message(i:i)) - 97 - s, 26) + 97)
      message(i:i) = ltr
    end if
  end do
end subroutine 
end program Caeser

File: Pascal/caeser.pas
program Caeser;
type 
  messages = string;
  values = integer;

var 
  csrString, originalMsg: messages;
  key: values;

function encrypt(var csrString: string; key: integer): string;
var 
  i: integer;
begin
  for i := 1 to Length(csrString) do 
    case csrString[i] of
      'A' .. 'Z' : csrString[i] := chr(ord('A') + (ord(csrString[i]) - ord('A') + key) mod 26);
      'a' .. 'z' : csrString[i] := chr(ord('a') + (ord(csrString[i]) - ord('a') + key) mod 26);
    end;
encrypt := csrString;
end;

function decrypt(var csrString: string; key: integer): string;
var 
  i: integer;
begin
  for i := 1 to Length(csrString) do 
    case csrString[i] of
      'A' .. 'Z' : csrString[i] := chr(ord('A') + (ord(csrString[i]) - ord('A') - key) mod 26);
      'a' .. 'z' : csrString[i] := chr(ord('a') + (ord(csrString[i]) - ord('a') - key) mod 26);
    end;
decrypt := csrString;
end;


begin
  writeln('Type your message: ');
  readln(csrString);
 
  writeln('Enter cipher: ');
  readln(key);

  originalMsg := csrString;

  writeln('Original message: ', csrString);
  encrypt(csrString, key);
  writeln('Encrypted message: ', csrString);
  decrypt(csrString, key);
  writeln('Decrypted message: ', csrString);
end.
